### PR TITLE
fix: Generate proper default string value for JavaScript

### LINKED ISSUE #5238: Default JavaScript value for string malformed
### Dafny version

4.5.0

### Code to produce this issue

```dafny
method Main() {
  var s: seq<char>;
  s := *;
  print "(", s, ")\n";
}
```

### Command to run and resulting output

```code
% dafny run test.dfy --target=js

Dafny program verifier finished with 1 verified, 0 errors
[stdin]:1122
        throw e
        ^

TypeError: _0_s.toVerbatimString is not a function
    at Function.Main ([stdin]:1202:35)
    at [stdin]:1211:53
    at Object.$module.HandleHaltExceptions ([stdin]:1116:7)
    at [stdin]:1211:8
    at Script.runInThisContext (node:vm:131:12)
    at Object.runInThisContext (node:vm:308:38)
    at node:internal/process/execution:81:19
    at [stdin]-wrapper:6:22
    at evalScript (node:internal/process/execution:80:60)
    at node:internal/main/eval_stdin:29:5
(
```

### What happened?

The JavaScript code generator has a special case for a `seq<char>` default value. Unfortunately, it gets it wrong.

### What type of operating system are you experiencing the problem on?

Mac

### PR DESCRIPTION
**Pull request by RustanLeino**
Fixes #5238

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Review by keyboardDrummer — APPROVED** [review]

File: Source/DafnyCore/Backends/JavaScript/JavaScriptCodeGenerator.cs [hunk 1]
```diff
@@ -974,7 +974,7 @@ protected class ClassWriter : IClassWriter {
         return $"{DafnyMultiSetClass}.Empty";
       } else if (xType is SeqType seq) {
         if (seq.Arg.IsCharType) {
-          return "''";
+          return "_dafny.Seq.UnicodeFromString('')";
```
Comment (fabiomadge) [inline Source/DafnyCore/Backends/JavaScript/JavaScriptCodeGenerator.cs:977]: Does this still support `--unicode-char=false`?

Comment (RustanLeino) [reply Source/DafnyCore/Backends/JavaScript/JavaScriptCodeGenerator.cs]: Good call. The answer happens to be "yes", and I really mean "happens to". I changed the code to treat the two cases separately.
In writing more test cases, I also ran across the known non-uniformity with the printing of empty strings in Python. I was able to remove this non-uniformity for the default string.

**Review by fabiomadge — APPROVED** [review]
